programs/cw-ics08-wasm-eth/src/state.rs

```diff
@@ -106,3 +106,24 @@ pub fn store_client_state(
 
     Ok(())
 }
+
+/// Store the eth client state
+/// # Errors
+/// Returns an error if the client state cannot be serialized into an Any
+// Note this is a function that will store a change to the internal eth client state
+// wrapped by the WasmClientState
+// It will replace the eth client state, while keeping the checksum and latest height
+// the same. Note if the latest height is changed, then the wasm client state
+// will be invalid.
+#[allow(clippy::module_name_repetitions)]
+pub fn store_eth_client_state(
+    storage: &mut dyn Storage,
+    eth_client_state: &EthClientState,
+) -> Result<(), ContractError> {
+    let mut wasm_client_state = get_wasm_client_state(storage)?;
+    wasm_client_state.data = serde_json::to_vec(eth_client_state)?;
+
+    store_client_state(storage, &wasm_client_state)?;
+
+    Ok(())
+}
```

Comment:
This function is not accurate since it doesn't update the `wasm_client_state.latest_height`.
It also assumes that `wasm_client_state` has been written already.
I doubt we need such a function

Reply:
i just needed it because i wanted to just store the eth client state in the wasm client state. but when i tried to use the existing function i needed to grab the wasm checksum
i actually think we can do without this. i can make this change

Reply:
Because there's no function to get the wasm client state.

Weirdly we have a function to get eth client state, and to set wasm client state. So the set/get are on different structs technically